ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs

```diff
@@ -261,6 +331,9 @@ instance Crypto c => ShowQuery (Query (ShelleyBlock c)) where
 getPParams :: SL.ShelleyState c -> SL.PParams
 getPParams = SL.esPp . SL.nesEs
 
+getProposedPPUpdates :: SL.ShelleyState c -> SL.ProposedPPUpdates c
```

Comment:
What's the proposed use of this? I'm tempted to say that in its current form it isn't terribly useful, since this gives no indication of when the proposal will come in or whether it's likely to do so...

Is https://github.com/input-output-hk/cardano-ledger-specs/issues/1337 likely to be useful here?

Reply:
I must admit that I'm not familiar with all these types. I wouldn't be surprised that we'll need to extend the `Query` type in the future. But that shouldn't be too difficult. This PR adds the necessary infrastructure (+ test infrastructure) so that future updates should be straightforward.